packages/admin-panel/src/utilities/userAccess.js

```diff
@@ -3,32 +3,35 @@ import { useUser } from '../api/queries';
 const BES_ADMIN_PERMISSION_GROUP = 'BES Admin';
 const VIZ_BUILDER_PERMISSION_GROUP = 'Viz Builder User';
 
-// Assume that if a user has any BES Admin access, they are an internal user,
-// to avoid having to check permissions for every country
-export const getHasBESAdminAccess = user => {
-  if (!user?.accessPolicy) return false;
-  return Object.keys(user.accessPolicy).some(countryCode =>
-    user.accessPolicy[countryCode].some(
-      permissionGroupName => permissionGroupName === BES_ADMIN_PERMISSION_GROUP,
-    ),
-  );
-};
+/**
+ * @param {string} permissionGroupName
+ * @returns {boolean | undefined} `true` if the user has the {@link permissionGroupName} permission
+ * group, `false` if they don’t, and `undefined` if the result is pending.
+ */
+export function useHasPermissionGroup(permissionGroupName) {
+  const { data: user, isFetched } = useUser();
+  if (!isFetched) return undefined;
+  if (!user?.accessPolicy) return false; // Should never happen
 
-export const getHasVizBuilderAccess = user => {
-  if (!user || !user.accessPolicy) return false;
-  return Object.keys(user.accessPolicy).some(countryCode =>
-    user.accessPolicy[countryCode].some(
-      permissionGroupName => permissionGroupName === VIZ_BUILDER_PERMISSION_GROUP,
-    ),
-  );
-};
+  // Permission groups the current user has access to for any country
+  const allPermissionGroups = new Set(Object.values(user.accessPolicy).flat());
+  return allPermissionGroups.has(permissionGroupName);
+}
 
 /**
- * @returns {boolean | undefined} `true` if the user has the BES Admin permission group, `false` if
- * they don’t, and `undefined` if the result is pending.
+ * @returns {boolean | undefined} `true` if the user has the {@link BES_ADMIN_PERMISSION_GROUP}
+ * permission group, `false` if they don’t, and `undefined` if the result is pending.
+ * @privateRemarks Assume that if a user has any BES Admin access, they are an internal user; and
```

Comment:
```suggestion
 * @privateRemarks You can assume that if a user has any BES Admin access, they are an internal user; and
```
Not as docsy, but because I'm so used to reading inline comments that describe what's coming up in the code that follows, this oriented me to expect that logic to exist inside the function. It's actually an instruction to the user, which is clear if you're used to reading jsdoc privateRemarks, I'm sure, but I'm not.

Reply:
Ah, I had moved the comment without interrogating it at all. Agree with your assessment here—thanks 🙏 

Updated in c8d082abe061ce12f12f8f7d57cc1eb215cb9efc